web/src/components/Accordion.tsx

```diff
@@ -34,18 +39,45 @@ export default function Accordion({
   setState: (isCollapsed: boolean) => void;
   isTopExpanding?: boolean;
 }) {
+  const { ref, height: observerHeight } = useResizeObserver<HTMLDivElement>();
+  const [renderChildren, setRenderChildren] = useState(!isCollapsed);
+
+  const [spring, api] = useSpring(
+    () => ({
+      height: isCollapsed ? 0 : observerHeight,
+      // eslint-disable-next-line unicorn/no-nested-ternary -- it interferes with prettier
+      rotate: isCollapsed ? (isTopExpanding ? -90 : 90) : isTopExpanding ? 90 : -90,
+    }),
+    [isCollapsed, isTopExpanding, observerHeight]
+  );
+
   const handleToggleCollapse = () => {
     onClick?.();
 
     isCollapsed && onOpen?.();
 
+    setRenderChildren(true);
+
+    api.start({
```

Comment:
I'm not super familiar with react-spring. Does it handle `api.stop()` on unmount for us, or do we need to call it in a clean up function?

Reply:
From what I've read in their documentation and some previous knowledge I think they handle that for us. But I'll try to double check this as well.

Reply:
Yeah I skimmed the docs briefly and didn't see anything, but thought I'd  double-check since you have more xp here 👍